app/ethWallet-geth.js

```diff
@@ -293,6 +295,40 @@ ipcMain.on('eth-wallet-get-keys-path', (e) => {
   e.sender.send('eth-wallet-keys-path', path.join(gethDataDir, 'keystore'))
 })
 
+ipcMain.on('eth-wallet-get-password-hash', (e) => {
+  fs.readFile(pwPath, 'utf8', (err, hash) => {
+    if (err || !hash) {
+      e.sender.send('eth-wallet-password-hash', null)
+    } else {
+      e.sender.send('eth-wallet-password-hash', hash)
+    }
+  })
+})
+
+ipcMain.on('eth-wallet-store-password-hash', (e, str) => {
+  bcrypt.genSalt(10, (err, salt) => {
+    if (err) {
+      return
+    }
+    bcrypt.hash(str, salt, (err, hash) => {
+      if (err) {
+        return
+      }
+      fs.writeFile(pwPath, hash)
+    })
+  })
+})
+
+ipcMain.on('eth-wallet-get-compare-password-hash', (e, str, hash) => {
+  bcrypt.compare(str, hash, (err, res) => {
+    if (err) {
+      e.sender.send('eth-wallet-compare-password-hash', false)
+    } else {
+      e.sender.send('eth-wallet-compare-password-hash', res)
+    }
+  })
+})
+
```

Comment:
In any case we are checking for `if (err)`, let's go ahead and output it via `console.error`